src/artemis/ispyb/store_in_ispyb.py

```diff
@@ -15,32 +15,37 @@ class StoreInIspyb:
 
     VISIT_PATH_REGEX = r".+/([a-zA-Z]{2}\d{4,5}-\d{1,3})/"
 
-    def __init__(self, ispyb_config, full_params: FullParameters):
+    def __init__(self, ispyb_config):
```

Comment:
I think passing the params in via the method makes a little more sense as the purpose of the object should be to store grid scan data in ispyb, rather than to store one specific data set.

For example, I think the `update_grid_scan_with_end_time_and_status` method as it currently is should be callable without needing to have a set of `FullParameters` available to create the object, and it should be possible to store multiple data sets without needing to create a new object each time.

Reply:
Ok then